numba/targets/builtins.py

```diff
@@ -49,6 +47,36 @@ def generic_is(context, builder, sig, args):
     else:
         return cgutils.false_bit
 
+
+@lower_builtin('==', types.Any, types.Any)
+def generic_eq(context, builder, sig, args):
+    """
+    Default implementation for `x == y` that fallback to `is`
```

Comment:
Doesn't `is` fallback to `==` for most types? This seems it could produce an infinite loop.